pkg/api/cached.go

```diff
@@ -285,16 +285,26 @@ func (s *Cached) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpublish
 
 	targetPath := req.GetTargetPath()
 	volumePath := path.Join(targetPath, "..")
+	upperDir := path.Join(volumePath, UPPER_DIR)
+	workDir := path.Join(volumePath, WORK_DIR)
+
+	// Check if we have anything to clean up
+	_, err := os.Stat(upperDir)
+	if err != nil && os.IsNotExist(err) {
+		return &csi.NodeUnpublishVolumeResponse{}, nil
+	}
+	_, err = os.Stat(workDir)
```

Comment:
If we crash halfway through this, the early return means we won't retry the other bits. Thoughts on just trying each piece individually each time and recovering from the its-already-done-error in each spot? I think it'd be good to be as defensive as possible in here.

Reply:
I do wonder if k8s doesn't just clean it all up anyhow but yeah we can be more defensive here.

Reply:
Ok, made the code more defensive.